src/java/org/apache/cassandra/utils/IFilterSerializer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.io.IOException;
+
+import org.apache.cassandra.io.util.DataOutputPlus;
+
+public interface IFilterSerializer
+{
```

Comment:
I didn't add `deserialize` method here because there is no flag or header from BF file to indicate the serializer type. They will continue to use `BloomFilterSerializer`.

Reply:
I think keeping `BloomFilter.serializer` accessible opens to misuse by directly invoking `serialize()`, while we should always use `getSerializer()`.

In order to overcome that and enforce usage of interface methods, I've pushed: https://github.com/datastax/cassandra/pull/1711/commits/92bb7e46a8ccdf5671caee7651858fa1f3e246a4, what do you think?

Reply:
good idea +1